pkgs/fc/sensuplugins/fc/sensuplugins/bgp_sessions.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+"""Check that FRR BGP sessions are up."""
+
+import argparse
+import json
+import subprocess
+import sys
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i",
+        "--interface",
+        dest="interfaces",
+        action="append",
+        default=[],
+        help="Check BGP session for named interface",
+    )
+    parser.add_argument(
+        "-c",
+        "--critical-age",
+        default=60,
+        metavar="SECONDS",
+        help="Report critical status if session not in established state for longer than SECONDS seconds",
+    )
+
+    args = parser.parse_args()
+
+    data = subprocess.check_output(["vtysh", "-c", "show bgp summary json"])
+    data = json.loads(data.decode("utf-8"))
+    data = data["ipv4Unicast"]["peers"]
+
+    oks = set()
+    errors = {}
+    warnings = {}
+    for iface in args.interfaces:
+        if iface not in data:
+            errors.update({iface: "no bgp session configured for interface"})
+            continue
+
+        if data[iface]["state"] == "Established":
+            oks.add(iface)
+            continue
+
+        uptime = int(data[iface]["peerUptimeMsec"] / 1000)
```

Comment:
What kind of `uptime` is this, uptime of the interface or of the bgp session? Depending on that, this can read to me like we're still issuing warnings or errors despite there being a session with a certain uptime. This is counterintuitive to me.

Reply:
Fixed, renamed to `age`.